module/zfs/vdev_label.c

```diff
@@ -208,14 +208,77 @@ vdev_label_write(zio_t *zio, vdev_t *vd, int l, void *buf, uint64_t offset,
 }
 
 /*
+ * Generate the nvlist representing this vdev's stats
+ */
+static void
+vdev_config_generate_stats(vdev_t *vd, nvlist_t *nv)
+{
+	vdev_stat_t *vs;
+	vdev_stat_ex_t *vsx;
+
+	vs = kmem_alloc(sizeof (*vs), KM_SLEEP);
+	ASSERT(vs);
+
+	vsx = kmem_alloc(sizeof (*vsx), KM_SLEEP);
+	ASSERT(vsx);
+
+	vdev_get_stats_ex(vd, vs, vsx);
+	fnvlist_add_uint64_array(nv, ZPOOL_CONFIG_VDEV_STATS,
+			(uint64_t *)vs, sizeof (*vs) / sizeof (uint64_t));
+
+	kmem_free(vs, sizeof (*vs));
+
+	/* Add extended stats nvlists */
+	fnvlist_add_uint64_array(nv, ZPOOL_CONFIG_QUEUE_ACTIVE,
+	    vsx->vsx_queue_active, ARRAY_SIZE(vsx->vsx_queue_active));
```

Comment:
It's not inconceivable additional queues might be added some day (or maybe removed?).  Using a uint64_t nvlist entry per-queue type would allow us to future proof this.

Have you given any thought to placing all of these new extended stats in to a new `ZPOOL_CONFIG_VDEV_STATS_EX` which is itself a nvlist.  You can add nvlists to nvlists with `fnvlist_add_nvlist()`.  This could make it easier to pass the extended stats around and allow them to be easily added/removed without serious user/kernel compatibility issues.